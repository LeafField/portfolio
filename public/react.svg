<svg width="113" height="100" viewBox="0 0 113 100" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="react 1" clip-path="url(#clip0_409_113)">
<g id="Group">
<path id="Vector" d="M92.8801 32.3788C91.7153 31.9779 90.5084 31.5985 89.2656 31.2398C89.4699 30.4062 89.6574 29.5834 89.824 28.7745C92.5601 15.4934 90.7712 4.79397 84.6625 1.27127C78.8051 -2.10653 69.2257 1.41537 59.5509 9.83499C58.6206 10.6446 57.6877 11.5018 56.7551 12.4002C56.1338 11.8058 55.5134 11.2314 54.8946 10.682C44.7552 1.67918 34.5921 -2.11473 28.4894 1.41816C22.6375 4.80576 20.9046 14.8642 23.3674 27.4509C23.6052 28.6666 23.8833 29.9081 24.1976 31.1704C22.7592 31.5788 21.3706 32.014 20.042 32.4771C8.15462 36.6216 0.562805 43.117 0.562805 49.8544C0.562805 56.8128 8.71262 63.7923 21.0942 68.0243C22.0713 68.3582 23.0852 68.674 24.1301 68.9738C23.7909 70.339 23.4959 71.6769 23.2483 72.9818C20.9 85.35 22.7339 95.1708 28.5701 98.537C34.5984 102.013 44.7157 98.4401 54.5673 89.8286C55.3459 89.1479 56.1274 88.426 56.9102 87.6693C57.9249 88.6465 58.938 89.5712 59.9457 90.4386C69.4881 98.6501 78.9128 101.966 84.7436 98.5904C90.766 95.1041 92.7232 84.5542 90.1822 71.7185C89.9881 70.7382 89.7623 69.7367 89.5087 68.7174C90.2192 68.5073 90.9167 68.2905 91.5968 68.0651C104.469 63.8003 112.844 56.9058 112.844 49.8544C112.844 43.0926 105.007 36.5534 92.8801 32.3788ZM90.0883 63.5121C89.4743 63.7153 88.8443 63.9118 88.2024 64.1023C86.7814 59.6037 84.8636 54.8201 82.5166 49.8778C84.7562 45.0532 86.5999 40.3307 87.9812 35.8613C89.1297 36.1937 90.2445 36.5442 91.3188 36.9142C101.709 40.491 108.047 45.7795 108.047 49.8544C108.047 54.1949 101.202 59.8295 90.0883 63.5121ZM85.4767 72.65C86.6003 78.3258 86.7608 83.4573 86.0165 87.4689C85.3478 91.0735 84.003 93.4768 82.3402 94.4393C78.8017 96.4875 71.2346 93.8252 63.0738 86.8026C62.1383 85.9975 61.196 85.1379 60.2506 84.2282C63.4144 80.768 66.5763 76.7455 69.6624 72.2781C75.0903 71.7964 80.2185 71.009 84.869 69.9339C85.098 70.8579 85.3015 71.764 85.4767 72.65ZM38.8418 94.0854C35.3847 95.3063 32.6312 95.3413 30.9668 94.3816C27.4251 92.3388 25.9528 84.4536 27.9612 73.8763C28.1912 72.665 28.465 71.4207 28.7806 70.1488C33.3801 71.1659 38.4707 71.8977 43.9116 72.3388C47.0184 76.7103 50.2717 80.7285 53.5502 84.2468C52.8339 84.9383 52.1204 85.5966 51.4109 86.217C47.0546 90.0249 42.6891 92.7265 38.8418 94.0854ZM22.6455 63.4855C17.1706 61.6142 12.6492 59.1822 9.55003 56.5282C6.7652 54.1433 5.3592 51.7756 5.3592 49.8544C5.3592 45.7659 11.4547 40.551 21.6209 37.0065C22.8544 36.5764 24.1457 36.1711 25.4857 35.7904C26.8907 40.3607 28.7338 45.1391 30.958 49.9711C28.705 54.8748 26.8358 59.7294 25.417 64.3533C24.4616 64.0787 23.5361 63.7897 22.6455 63.4855ZM28.0747 26.5297C25.9646 15.7458 27.366 7.61077 30.8925 5.56942C34.6488 3.39476 42.9551 6.49536 51.7097 14.2688C52.2693 14.7657 52.8311 15.2857 53.3945 15.8237C50.1322 19.3267 46.9087 23.3146 43.8295 27.6603C38.5488 28.1498 33.494 28.936 28.8465 29.9889C28.5543 28.8135 28.2955 27.6587 28.0747 26.5297ZM76.5075 38.4891C75.3965 36.57 74.2559 34.6963 73.0931 32.8738C76.6756 33.3267 80.1079 33.9279 83.3313 34.6636C82.3635 37.765 81.1574 41.0077 79.7376 44.3319C78.7191 42.3983 77.6419 40.4487 76.5075 38.4891ZM56.7567 19.2517C58.9691 21.6486 61.1848 24.3247 63.364 27.2282C61.168 27.1245 58.9448 27.0706 56.7032 27.0706C54.4827 27.0706 52.2759 27.1233 50.0934 27.2252C52.2749 24.3487 54.5097 21.6762 56.7567 19.2517ZM36.8794 38.5222C35.7696 40.4467 34.7124 42.3857 33.7099 44.3283C32.3131 41.0157 31.1181 37.7582 30.142 34.6128C33.3455 33.8959 36.7617 33.3097 40.3214 32.8654C39.1422 34.7047 37.9922 36.5926 36.8794 38.5222ZM40.4239 67.1855C36.7461 66.7752 33.2786 66.2192 30.0759 65.5221C31.0676 62.3205 32.2889 58.9935 33.7153 55.6095C34.7204 57.5507 35.7818 59.4905 36.8984 61.4192C38.0358 63.384 39.2144 65.3086 40.4239 67.1855ZM56.8894 80.7952C54.616 78.3424 52.3484 75.6291 50.1338 72.7106C52.2837 72.7949 54.4755 72.8381 56.7032 72.8381C58.9917 72.8381 61.2541 72.7865 63.4809 72.6874C61.2945 75.6585 59.0869 78.3759 56.8894 80.7952ZM79.7834 55.437C81.2841 58.8576 82.5492 62.1672 83.5537 65.3142C80.2982 66.0569 76.7833 66.6551 73.0799 67.0995C74.2455 65.2525 75.3961 63.3466 76.5267 61.3862C77.6705 59.4028 78.7565 57.4164 79.7834 55.437ZM72.3718 58.9899C70.6167 62.0331 68.8148 64.9383 66.9855 67.6805C63.6536 67.9188 60.2115 68.0415 56.7032 68.0415C53.209 68.0415 49.8102 67.9329 46.5374 67.7205C44.6343 64.9423 42.7932 62.0285 41.0493 59.0161C39.3099 56.0118 37.7094 52.9822 36.2589 49.9695C37.709 46.9498 39.3057 43.9168 41.0351 40.9179L41.0347 40.9185C42.7686 37.9117 44.5936 35.0121 46.479 32.2513C49.8182 31.9989 53.2424 31.8672 56.703 31.8672C60.1791 31.8672 63.6079 31.9999 66.9461 32.2545C68.8028 34.9951 70.6153 37.8853 72.356 40.8925C74.1166 43.9336 75.7339 46.9464 77.1964 49.9013C75.7382 52.907 74.1246 55.9503 72.3718 58.9899ZM82.2666 5.42672C86.0265 7.59498 87.4887 16.3396 85.1264 27.8064C84.9757 28.5381 84.806 29.2834 84.6217 30.0386C79.9634 28.9638 74.905 28.164 69.609 27.6671C66.5238 23.2737 63.3267 19.2795 60.1175 15.8215C60.9803 14.9915 61.8419 14.2001 62.6999 13.4532C70.9892 6.23934 78.7367 3.39116 82.2666 5.42672ZM56.7032 39.8281C62.2404 39.8281 66.7294 44.3169 66.7294 49.8544C66.7294 55.3917 62.2404 59.8807 56.7032 59.8807C51.1659 59.8807 46.6769 55.3917 46.6769 49.8544C46.6769 44.3169 51.1659 39.8281 56.7032 39.8281Z" fill="#00D8FF"/>
</g>
</g>
<defs>
<clipPath id="clip0_409_113">
<rect width="112.281" height="100" fill="white" transform="translate(0.562805)"/>
</clipPath>
</defs>
</svg>
