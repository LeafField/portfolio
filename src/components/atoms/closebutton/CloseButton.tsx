import React from "react";
import { roboto } from "../../../lib/font";
import useStore from "../../../store";

const CloseButton = () => {
  const { showModal } = useStore();

  return (
    <button
      className={`flex items-center justify-center border border-mainColor/30 px-3 py-1 text-main-textColor transition-colors duration-300 hover:border-mainColor ${roboto.className}`}
      onClick={() => showModal({ show: false })}
    >
      Close
    </button>
  );
};

export default CloseButton;
