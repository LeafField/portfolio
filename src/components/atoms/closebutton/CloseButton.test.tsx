import { render } from "@testing-library/react";
import CloseButton from "./CloseButton";

describe("CloseButtonの単体テスト", () => {
  it("スナップショットテスト", () => {
    const { container } = render(<CloseButton />);
    expect(container).toMatchSnapshot();
  });
});
