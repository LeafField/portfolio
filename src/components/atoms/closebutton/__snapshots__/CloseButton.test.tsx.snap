// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButtonの単体テスト スナップショットテスト 1`] = `
<div>
  <button
    class="flex items-center justify-center border border-mainColor/30 px-3 py-1 text-main-textColor transition-colors duration-300 hover:border-mainColor className"
  >
    Close
  </button>
</div>
`;
